chaincode/ledger_db.go

```diff
@@ -480,31 +480,49 @@ func (db *LedgerDB) AddTupleEvent(tupleKey string) error {
 		}
 		out := outputTraintuple{}
 		out.Fill(db, tuple, tupleKey)
-		db.tuplesEvent.Traintuples = append(db.tuplesEvent.Traintuples, out)
+		db.event.Traintuples = append(db.event.Traintuples, out)
 	case CompositeTraintupleType:
 		tuple, err := db.GetCompositeTraintuple(tupleKey)
 		if err != nil {
 			return err
 		}
 		out := outputCompositeTraintuple{}
 		out.Fill(db, tuple, tupleKey)
-		db.tuplesEvent.CompositeTraintuples = append(db.tuplesEvent.CompositeTraintuples, out)
+		db.event.CompositeTraintuples = append(db.event.CompositeTraintuples, out)
 	case AggregatetupleType:
 		tuple, err := db.GetAggregatetuple(tupleKey)
 		if err != nil {
 			return err
 		}
 		out := outputAggregatetuple{}
 		out.Fill(db, tuple, tupleKey)
-		db.tuplesEvent.Aggregatetuples = append(db.tuplesEvent.Aggregatetuples, out)
+		db.event.Aggregatetuples = append(db.event.Aggregatetuples, out)
 	case TesttupleType:
 		tuple, err := db.GetTesttuple(tupleKey)
 		if err != nil {
 			return err
 		}
 		out := outputTesttuple{}
 		out.Fill(db, tupleKey, tuple)
-		db.tuplesEvent.Testtuples = append(db.tuplesEvent.Testtuples, out)
+		db.event.Testtuples = append(db.event.Testtuples, out)
 	}
 	return nil
 }
+
+// AddComputePlanEvent add the compute plan matching the ID to the event struct
+func (db *LedgerDB) AddComputePlanEvent(ComputePlanID, status string) error {
+	if !stringInSlice(status, []string{StatusCanceled, StatusFailed, StatusDone}) {
+		return nil
+	}
+	cp := eventComputePlan{
+		ComputePlanID: ComputePlanID,
+		Status:        status,
+	}
+	algokeys, err := db.GetIndexKeys("algo~computeplanid~key", []string{"algo", ComputePlanID})
+	if err != nil {
+		return err
+	}
+	cp.AlgoKeys = algokeys
```

Comment:
is it possible to have more than one algo for a compute plan?

Reply:
Yes it was implemented directly after the creation of composite train tuple and aggregate tuple since at least there was one algo per tuple type.